tests/integration/integration-test.py

```diff
@@ -62,6 +63,110 @@ def test_get_notes_path(self, endpoint='/notes'):
+            schema = utils.get_resource_schema(self, 'Error')
+            utils.check_schema(self, response, schema)
+
+    # /notes?studentId=111111111&creatorId
+    def query_creator_id(self, student_id, creator_ids, endpoint='/notes'):
+        for creator_id in creator_ids:
+            params = {'studentId': student_id,'creatorId': creator_id}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            schema = utils.get_resource_schema(self, 'NoteResource')
+            utils.check_schema(self, response, schema)
+            # Validating the creatorId requested is the same creatorId received
+            for resource in response.json()['data']:
+                actual_creator_id = resource['attributes']['creatorId']
+                self.assertEqual(actual_creator_id, creator_id)
+
+    # /notes?studentId=111111111&contextTypes
+    def query_context_types(self, student_id, context_types, endpoint='/notes'):
+        for context_type in context_types:
+            params = {'studentId': student_id,'contextTypes': context_type}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            schema = utils.get_resource_schema(self, 'NoteResource')
+            utils.check_schema(self, response, schema)
+            # Validating the contextTypes requested is the same contextTypes received
+            for resource in response.json()['data']:
+                actual_context_type = resource['attributes']['context']['contextType']
+                self.assertEqual(actual_context_type, context_type)
+
+    # /notes?studentId=111111111&q=
+    def query_string_search(self, student_id, notes, endpoint='/notes'):
+        for note in notes:
+            q = random.choice(note.split(' '))
+            params = {'studentId': student_id, 'q': q}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            schema = utils.get_resource_schema(self, 'NoteResource')
+            utils.check_schema(self, response, schema)
+            # Validating the search word requested exist in the notes received
+            for resource in response.json()['data']:
+                actual_note = resource['attributes']['note']
+                self.assertIn(q, actual_note)
+
+    # /notes?studentId=111111111&sources=
+    def query_source(self, student_id, sources, endpoint='/notes'):
+        for source in sources:
+            params = {'studentId': student_id, 'sources': source}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            schema = utils.get_resource_schema(self, 'NoteResource')
+            utils.check_schema(self, response, schema)
+            # Validating the source requested is the same source 
+            for resource in response.json()['data']:
+                actual_source = resource['attributes']['source']
+                self.assertEqual(actual_source, source)
```

Comment:
`sources` is also csv supported filter. Also, this parameter can also cause 400 bad requests. Can you add logic to cover these scenarios?